tests/new_api_test.cpp

```diff
@@ -1398,6 +1441,29 @@ bool register_tests(const nodes_data *setup) {
 		ELLIPTICS_TEST_CASE(test_write_cas, use_session(n, {}, 0, ioflags));
 
 		ELLIPTICS_TEST_CASE(test_write_to_readonly_backend, use_session(n, {}, 0, ioflags), setup);
+
+		// Test key remove with and without CAS timestamp
+		record.key = {in_cache ? "cache_remove_key" : "remove_key"};
+		for (int flags : {DNET_IO_FLAGS_CAS_TIMESTAMP, 0}) {
+			dnet_current_time(&record.timestamp);
+			ELLIPTICS_TEST_CASE(test_write, use_session(n, {}, 0, ioflags), record);
+
+			// if test uses CAS timestamp, then key removal with older session timestamp
+			// must return -EBADFD
+			record.timestamp.tsec -= 1;
```

Comment:
Why is `record.timestamp` used for specifying timestamp for a session? `test_remove` and `test_remove_corrupted` should accepts timestamp instead.